tgui/packages/tgui/interfaces/CharacterPreferences/index.tsx

```diff
@@ -0,0 +1,183 @@
+                      <Button icon="chevron-right" onClick={() => act('rotate-counter-clockwise')} />
+                    </Box>
+                  </Section>
+                </Stack.Item>
+              </Stack>
+            ) : (
+              <Section scrollable fill>
+                {menu === CharacterPreferencesTabKeys.GameSettings && <GameSettingsTab />}
+                {menu === CharacterPreferencesTabKeys.Saves && <SavesTab />}
+              </Section>
+            )}
+          </Stack.Item>
+          <Stack.Item>
+            <Section>
+              <Button.Confirm content="Reset All" onClick={() => act('reset')} />
+            </Section>
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SavesAndProfile = (_props, context) => {
+  const { act, data } = useBackend<CharacterPreferencesData>(context);
+
+  const activeProfileIndex = data.profiles.findIndex((p) => p.active);
+
+  return (
+    <Stack vertical>
+      <Stack.Item>
+        <Stack>
+          {data.profiles.map((profile, index) => (
+            <Stack.Item key={index} ml={index === 0 ? '0' : '5px'} grow="1">
+              <Profile profile={profile} index={index} />
+            </Stack.Item>
+          ))}
+        </Stack>
+      </Stack.Item>
+      <Stack.Item>
+        <Section>
+          <LabeledList>
+            <LabeledList.Item
+              label="Profile Name"
+              buttons={
+                activeProfileIndex > -1 ? (
+                  <>
+                    <Button onClick={() => act('load', { index: activeProfileIndex + 1 })}>Reload</Button>
+                    {' - '}
+                    <Button onClick={() => act('save', { index: activeProfileIndex + 1 })}>Save</Button>
+                  </>
+                ) : null
+              }>
+              <Button onClick={() => act('update', { profileName: 1 })}>
+                {data.profileName ? data.profileName : <Box italic>None</Box>}
+              </Button>
+            </LabeledList.Item>
+          </LabeledList>
+          {!!data.profileModified && (
+            <>
+              <Divider />
+              <NoticeBox danger>
+                <Stack>
+                  <Stack.Item align="center">
+                    <Icon name="exclamation-triangle" />
+                  </Stack.Item>
+                  <Stack.Item>
+                    <Box>You may have unsaved changes!</Box>
+                    <Box>Any unsaved changes will take effect for this round only.</Box>
+                  </Stack.Item>
+                </Stack>
+              </NoticeBox>
+            </>
+          )}
+        </Section>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+type ProfileProps = {
+  index: number,
+  profile: CharacterPreferencesProfile
+}
+
+const Profile = (
+  { profile, index } : ProfileProps,
+  context: any
+) => {
+  const { act } = useBackend<CharacterPreferencesData>(context);
```

Comment:
(Untested suggestion below)
```suggestion
const Profile = (props: ProfileProps, context: any) => {
  const {
    index,
    profile,
  } = props;
  const { act } = useBackend<CharacterPreferencesData>(context);
```